### PR TITLE
V22F-656 exit abnormal shut down

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Throws an exception if it cannot connect to speed dreams

### CODE REVIEW

File: SDALib-CPP/SDALibTests/SDADriverTest.cpp [hunk 1]
```diff
@@ -158,9 +167,29 @@ void WaitTest(int p_waitAmount, int p_situations)
...
 END_TEST_COMBINATORIAL2(WaitTest, waitAmounts, 2, situationAmounts, 3)
+
+/// @brief Tests the timeout of the driver.
+TEST(DriverTests, TimeoutTests)
+{
+    SDADriverMock driver;
+    ASSERT_DURATION_LE(45, ASSERT_THROW(driver.Run(), std::exception));
+}
+
+TEST(DriverTests, BreakingConnectionTest)
```
Comment (shoekzema) [inline SDALib-CPP/SDALibTests/SDADriverTest.cpp:185]: missing documentation

File: libraries/include/Socket.h [hunk 2]
```diff
@@ -20,10 +20,11 @@
...
 
-#define WSA_ERROR           -1
-#define IPCLIB_SERVER_ERROR 1
-#define IPCLIB_CLIENT_ERROR 2
-#define IPCLIB_SUCCEED      0
+#define WSA_ERROR            -1
+#define IPCLIB_SERVER_ERROR  1
+#define IPCLIB_CLIENT_ERROR  2
+#define IPCLIB_SUCCEED       0
+#define IPCLIB_RECEIVE_ERROR 3
```
Comment (shoekzema) [inline libraries/include/Socket.h:27]:
```suggestion
#define IPCLIB_SUCCEED       0
#define IPCLIB_SERVER_ERROR  1
#define IPCLIB_CLIENT_ERROR  2
#define IPCLIB_RECEIVE_ERROR 3
```

**Review by shoekzema — CHANGES_REQUESTED** [review]

**Review by shoekzema — APPROVED** [review]

File: SDALib-CPP/SDALib-CPP/SDADriver.hpp [hunk 3]
```diff
@@ -103,14 +109,15 @@ class SDALIB_EXPORT AIInterface
         msgpack::sbuffer sbuffer;
         msgpack::pack(sbuffer, order);
 
-        m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE);  // receive reply
+        IPC_OK(m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE), "Failed to receive message from server");  // receive reply
```
Comment (noutheesink) [inline SDALib-CPP/SDALib-CPP/SDADriver.hpp:112]:
```suggestion
        IPC_OK(m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE), "[SDA] Failed to receive message from server");  // receive reply
```

File: SDALib-CPP/SDALib-CPP/SDADriver.hpp [hunk 4]
```diff
@@ -103,14 +109,15 @@ class SDALIB_EXPORT AIInterface
...
         if (m_buffer[0] != 'O' || m_buffer[1] != 'K') throw std::exception("Server send wrong reply");
 
         sbufferCopy(sbuffer, m_buffer, SDA_BUFFER_SIZE);
 
         m_client.ReceiveDataAsync();
         IPC_OK(m_client.SendData(m_buffer, sbuffer.size()), "[SDA] Could not send order data");
-        m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE);
+
+        IPC_OK(m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE), "Failed to receive message from server");
```
Comment (noutheesink) [inline SDALib-CPP/SDALib-CPP/SDADriver.hpp:120]:
```suggestion
        IPC_OK(m_client.AwaitData(m_buffer, SDA_BUFFER_SIZE), "[SDA] Failed to receive message from server");
```

**Review by noutheesink — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]

**Pull request merged**